google-cloud-datastore/src/main/java/com/google/cloud/datastore/DatastoreException.java

```diff
@@ -38,13 +38,18 @@
   private static final long serialVersionUID = 2663750991205874435L;
```

Comment:
If you get DEADLINE_EXCEEDED, you don't know if the underlying commit was triggered or not (or whether it succeeded).

If you get UNAVAILABLE, I believe it is safe to retry the commit (we should only return that before we trigger the underlying commit). UNAVAILABLE can also be returned client-side if the HTTP request fails.